src/dump.rs

```diff
@@ -0,0 +1,105 @@
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct RustDoc {
+    deps: bool,
+    silence: bool,
+}
+
+impl RustDoc {
```

Comment:
Consider tweaking the name here to communicate that this is a type that can _generate_ rustdoc JSON, rather than a type that represents _already-generated_ rustdoc JSON?

Otherwise, it feels strange to see what appears to be the "current" rustdoc JSON instance being given to the baseline rustdoc JSON generator.

Reply:
Modeling off of `cargo_metadata`, calling this `RustDocCommand` seems appropriate.